presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -1484,4 +1493,59 @@ protocol::NodeStatus PrestoServer::fetchNodeStatus() {
   return nodeStatus;
 }
 
+void PrestoServer::convertToVeloxPlan(
+    proxygen::HTTPMessage* message,
+    proxygen::ResponseHandler* downstream,
+    const std::vector<std::unique_ptr<folly::IOBuf>>& body) {
+  protocol::ExecutionFailureInfo failure;
+  try {
+    auto headers = message->getHeaders();
+
+    std::string planFragmentJson = util::extractMessageBody(body);
+    protocol::PlanFragment planFragment = json::parse(planFragmentJson);
+
+    auto queryCtx = core::QueryCtx::create();
+    VeloxInteractiveQueryPlanConverter converter(queryCtx.get(), pool_.get());
```

Comment:
We could have a local Memory Pool for this code. Can avoid using the global server pool.

Reply:
Just curious to learn more about the memory pool hierarchy and why you shouldn't use the global pool (and the comment [here](https://github.com/facebookincubator/velox/blob/main/velox/common/memory/MemoryPool.h#L66-L112) was pretty helpful).  Is the purpose of the memory pool hierarchy so that we can track where the memory is going more granularly (how much memory of the query memory is used by sub-pool x vs. sub-pool y)?

Reply:
The MemoryPool is used for accounting memory used by the query. You are right, it tracks memory getting more granular: from query -> task -> PlanNode -> operator.

The VeloxPlanConversion code is only doing memory allocations for constant folding.

My general thought is to not keep adding these small allocations in the global root pool. We can add a leaf pool specifically for these plan conversions and use that throughout.

So add a new class variable in PrestoServer called planConversionPool maybe

`std::shared_ptr<memory::MemoryPool> nativeWorkerPool_;`
Construct it once in the PrestoServer initialization by doing :

```
nativeWorkerPool_ =
      velox::memory::MemoryManager::getInstance()->addLeafPool("PrestoNativeWorker");
```

And use nativeWorkerPool_.get() instead of pool_.get() in this code.

Reply:
Ok, I can try this. Just wondering though, does creating a leaf pool have enough cost that we would want to pre-construct the child like this, or could we just create it in the scope that it is used, like when the plan conversion endpoint is called?

Reply:
done

Reply:
I don't have a great answer about the cost of the leaf pool. Working with a single pool seemed like the more conservative option without any downside.